src/platform/packages/shared/kbn-scout/src/playwright/fixtures/scope/worker/api_key.ts

```diff
@@ -77,63 +127,67 @@ export const requestAuthFixture = coreWorkerFixtures.extend<
+                    `Is '${predefinedRole}' a predefined role? (e.g., admin, viewer, editor) → Add the role descriptor to ${defaultRoles.rolesFilePath} to enable it for testing.`,
+                  ].join('\n\n');
+                  throw new Error(errorMessage);
                 }
-                log.debug(
-                  `Creating API key for ${role} with privileges: ${JSON.stringify(roleDescriptor)}`
-                );
-                return { [role]: roleDescriptor };
+                return { [predefinedRole]: roleDescriptor };
               })();
+        return await createApiKeyWithAdminCredentials(predefinedRole, roleDescriptors);
+      };
 
-        const payload = createApiKeyPayload(apiKeyName, role, roleDescriptors);
-        const response = await apiClient.post('internal/security/api_key', {
-          headers: {
-            'kbn-xsrf': 'some-xsrf-token',
-            'x-elastic-internal-origin': 'kibana',
-            ...adminCookieHeader,
-          },
-          body: payload,
-          responseType: 'json',
-        });
-
-        if (response.statusCode !== 200) {
-          throw new Error(
-            `Failed to create API key for '${role}' role with response text: ${response.statusMessage}`
-          );
-        }
+      const getApiKeyForCustomRole = async (
+        roleDescriptor: KibanaRole | ElasticsearchRoleDescriptor
+      ): Promise<RoleApiCredentials> => {
+        isCustomRoleCreated = true;
 
-        const apiKey = response.body as ApiKey;
-        const apiKeyHeader = { Authorization: `ApiKey ${apiKey.encoded}` };
+        try {
+          await samlAuth.setCustomRole(roleDescriptor);
 
-        log.info(`Created API key for role: [${role}]`);
-        generatedApiKeys.push(apiKey);
-        return { apiKey, apiKeyHeader };
+          const result = await createApiKeyWithAdminCredentials(samlAuth.customRoleName, {
+            [samlAuth.customRoleName]: roleDescriptor,
+          });
+          return result;
+        } catch (error: any) {
+          log.error(`Failed to create API key for custom role: ${error.message}`);
+          throw error;
+        }
       };
 
-      await use({ getApiKey });
+      await use({ getApiKey, getApiKeyForCustomRole });
 
       // Invalidate all API Keys after tests
       await measurePerformanceAsync(log, `Delete all API Keys`, async () => {
         log.debug(`Delete all API Keys`);
         return invalidateApiKeys(generatedApiKeys);
       });
+
+      if (isCustomRoleCreated) {
+        log.debug(`Deleting custom role with name ${samlAuth.customRoleName}`);
+        try {
+          await esClient.security.deleteRole({ name: samlAuth.customRoleName });
+          log.info(`Deleted ${samlAuth.customRoleName} custom role`);
+        } catch (error: any) {
+          log.error(`Failed to delete custom role ${samlAuth.customRoleName}: ${error.message}`);
+        }
```

Comment:
@csr I have the same question here if we can continue while role deletion failed in the previous test